core/CronArchive.php

```diff
@@ -262,6 +267,8 @@ public function __construct($processNewSegmentsFrom = null, LoggerInterface $log
 
         $processNewSegmentsFrom = $processNewSegmentsFrom ?: StaticContainer::get('ini.General.process_new_segments_from');
         $this->segmentArchivingRequestUrlProvider = new SegmentArchivingRequestUrlProvider($processNewSegmentsFrom);
+
+        $this->invalidator = StaticContainer::get('Piwik\Archive\ArchiveInvalidator');
```

Comment:
It's closer to the constructor injection pattern, but the hard part in removing StaticContainer isn't moving the code to the constructor, but making injected classes immutable. So more habit for me at this point.